iOS-ObjC/MSALiOSobjC/ViewController.m

```diff
@@ -0,0 +1,278 @@
+     to sign out a user from this application.
+     */
+    
+    - (IBAction)signOut{
+        
+        MSALAccount *account = [self currentAccount];
+        if (account == nil){
+            return;
+        }
+        
+        /**
+         Removes all tokens from the cache for this application for the provided account
+         
+         - account:    The account to remove from the cache
+         */
+        
+        [application removeAccount:account error:nil];
+        _signout.enabled = false;
+        [_resultTextView setText:@""];
+    }
+    
+    // MARK: Acquiring and using token
+    
+    /**
+     This will invoke the authorization flow.
+     */
+    
+    - (IBAction)callGraphAction: (id)sender{
+        
+        MSALAccount *msalAccount = [self currentAccount];
+        // We check to see if we have a current logged in account.
+        // If we don't, then we need to sign someone in.
+        if (msalAccount == nil){
+            [self acquireTokenInteractively];
+            return;
+        }
+        
+        [self acquireTokenSilently:msalAccount];
+    }
+    
+    -(void)acquireTokenSilently: (MSALAccount*) account {
+        /**
+         
+         Acquire a token for an existing account silently
+         
+         - forScopes:           Permissions you want included in the access token received
+         in the result in the completionBlock. Not all scopes are
+         guaranteed to be included in the access token returned.
+         - account:             An account object that we retrieved from the application object before that the
+         authentication flow will be locked down to.
+         - completionBlock:     The completion block that will be called when the authentication
+         flow completes, or encounters an error.
+         */
+        NSArray *scopes = @[@"https://graph.microsoft.com/user.read"];
+        
+        MSALSilentTokenParameters *parameters = [[MSALSilentTokenParameters alloc] initWithScopes:scopes account:account];
+        
+        // interactionRequired means we need to ask the user to sign-in. This usually happens
+        // when the user's Refresh Token is expired or if the user has changed their password
+        // among other possible reasons.
+        
+        [application acquireTokenSilentWithParameters:parameters completionBlock:^(MSALResult *result, NSError *error)
+         {
+             if (error != nil){
+                 if (error.domain == MSALErrorDomain) {
+                     if (error.code == MSALErrorInteractionRequired){
+                         dispatch_async(dispatch_get_main_queue(), ^{
+                             [self acquireTokenInteractively];
+                             return;
+                         });
+                     }
+                 }
+             }
+             
+             [self updateResultView:result];
+             dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
please have helper methods for all UI updates like in Swift sample

Reply:
updated